src/rngs/os.rs

```diff
@@ -46,15 +45,28 @@ use rand_core::{CryptoRng, RngCore, Error, impls};
 /// - Redox: reads from `rand:` device.
 /// - CloudABI: calls `random_get`.
 /// - Other Unix-like systems: reads directly from `/dev/urandom`.
-///   Note: many Unix systems provide `/dev/random` as well as `/dev/urandom`.
-///   On all modern systems these two interfaces offer identical quality, with
-///   the difference that on some systems `/dev/random` may block. This is a
-///   dated design, and `/dev/urandom` is preferred by cryptography experts. [1]
 ///
-/// [1] See [Myths about urandom](https://www.2uo.de/myths-about-urandom/).
+/// ## Notes on Unix `/dev/urandom`
+///
+/// Many Unix systems provide `/dev/random` as well as `/dev/urandom`. On all
+/// modern systems these two interfaces offer identical quality, with the
+/// difference that on some systems `/dev/random` may block. This is a dated
+/// design, and `/dev/urandom` is preferred by cryptography experts.
+/// See [Myths about urandom](https://www.2uo.de/myths-about-urandom/).
+///
+/// On some systems reading from `/dev/urandom` “may return data prior to the
+/// entropy pool being initialized”. I.e., early in the boot process, and
+/// especially on virtual machines, `/dev/urandom` may return data that is less
+/// random.
+///
+/// As a countermeasure we try to do a single read from `/dev/random` in
```

Comment:
I know you like short paragraphs, but this countermeasure is directly related to the above problem and should be part of the same paragraph. This is more a problem here than where the text was before.